CMakeLists.txt

```diff
@@ -297,8 +297,8 @@ endif()
 # Just use -lz and let Xcode figure it out
 # Emscripten does provide Zlib, but it doesn't work with find_package and is handled specially
 if(NOT APPLE AND NOT EMSCRIPTEN AND NOT TARGET ZLIB::ZLIB)
-    if(WIN32)
-        realm_acquire_dependency(zlib ${DEP_WIN32_ZLIB_VERSION} ZLIB_CMAKE_INCLUDE_FILE)
+    if(WIN32 OR (CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_CROSSCOMPILING))
```

Comment:
Why not have the toolchain files set a var like `USING_REALM_LINUX_TOOLCHAIN`, rather than relying on the somewhat odd semantics of `CMAKE_SYSTEM_NAME` and `CMAKE_CROSSCOMPILING`? I _think_ that is what you are really trying to check for here, and wouldn't want to use this logic just because someone happens to try to compile core using a cross-compiler.